Tela de Transactions

TTOExtreme opened the pull request:
-- Adicionado tabela com filtro de Colunas (Hard Coded)
-- Adicionado item para acesso a pagina de Transactions no Dropdown do User
-- WIP Criar icone da tela de Transactions
-- WIP Acertar a comunicação com a api
-- NTI Criar Micro Router para redirecionar requests do subdominio no modo DEV (usar uma mesma porta Ex 80 e checar se é a api > 3001 ou a root >3000)

Commit:
WIP Tela de Transactions

-- WIP Filtro de transações
-- WIP Request da lista pela API
-- NTI Request para um subdomínio
-- NTI Salvar a lista na Store e verificar somente a ultima para checar se tem atualizado

Commit:
WIP Tela de Transactions

-- WIP Filtro de transações
-- WIP Request da lista pela API
-- NTI Request para um subdomínio
-- NTI Salvar a lista na Store e verificar somente a ultima para checar se tem atualizado

Commit:
Filtro de tabela funcional

-- Alterado layout da pagina para permitir informações em tela cheia
-- Adicionado Filtro na tabela de transactions
-- WIP Criar icone da tela de Transactions
-- WIP Corrigir erro no acesso da API pelo em modo DEV

Commit:
Filtro de tabela funcional

-- Alterado layout da pagina para permitir informações em tela cheia
-- Adicionado Filtro na tabela de transactions
-- WIP Criar icone da tela de Transactions
-- WIP Corrigir erro no acesso da API pelo em modo DEV

Commit:
Criado serviço de roteamento para o modo DEV

-- Serviço de roteamento de subdomínios para o modo DEV, redirecionando o dominio principal para a porta 3002 (handler da pagina web) e o subdominio api para a porta 3001

Commit:
Criado serviço de roteamento para o modo DEV

-- Serviço de roteamento de subdomínios para o modo DEV, redirecionando o dominio principal para a porta 3002 (handler da pagina web) e o subdominio api para a porta 3001

Commit:
Coleta de dados do servidor funcionando

-- Adicionado rota de login na api.../user/login (posivel alteração para  /)
-- DONE Salvar a lista na Store e verificar somente a ultima para checar se tem atualizado
-- DONE Request da lista pela API

Commit:
Coleta de dados do servidor funcionando

-- Adicionado rota de login na api.../user/login (posivel alteração para  /)
-- DONE Salvar a lista na Store e verificar somente a ultima para checar se tem atualizado
-- DONE Request da lista pela API

GCastilho commented on src/server/api/v1/user.ts:
```diff
@@ -89,6 +108,38 @@ router.get('/transactions/:opid', async (req, res) => {
 	}
 })
 
+/**
+ * Retorna a ultima transaction do usuario
+ */
+router.get('/lasttransaction', async (req, res) => {
+	/** Filtro de transações por currency */
+	const currency = CurrencyApi.currencies.find(currency => currency === req.query.currency)
+	/** Filtro da query do mongo */
+	const query = currency ? { user: req.user?.id, currency } : { user: req.user?.id }
+	/**
+	 * retorna a ultima transaction
+	 */
+	const txs = await Transaction.find(query, null, {
+		sort: { timestamp: -1 },
+		limit: 1,
+		skip:0
+	})
+	
+	const formattedTransactions = txs.map(tx => ({
+		opid:          tx._id.toHexString(),
+		status:        tx.status,
+		currency:      tx.currency,
+		txid:          tx.txid,
+		account:       tx.account,
+		amount:        tx.amount.toFullString(),
+		type:          tx.type,
+		confirmations: tx.confirmations,
+		timestamp:     tx.timestamp.getTime()
+	}))
+	res.send(formattedTransactions)
+})
```
Isso daqui é redundante pq essa informação pode ser pega na /transactions

GCastilho commented on src/server/root/login.js:
```diff
@@ -41,8 +41,9 @@ Router.post('/', function(req, res) {
 		 * @todo cookie ter tempo de expiração
+		 *
 		 */
-		res.cookie('sessionId', session.sessionId, { httpOnly: true })
+		res.cookie('sessionId', session.sessionId,)
```
Não, o cookie tem que ser HTTP Only

GCastilho commented on src/server/api/v1/user.ts:
```diff
@@ -17,6 +35,7 @@ router.use(async (req, res, next) => {
 	try {
 		if (!req.cookies.sessionId) throw 'Cookie Not Found'
 		req.user = await UserApi.findUser.byCookie(req.cookies.sessionId)
+		res.cookie('sessionId', req.cookies.sessionId,)
```
Pq vc está setando o cookie do usuário na função que checa se ele está logado?

TTOExtreme replied:
utilizava para adicionar um cookie com o subdominio api

GCastilho commented on src/server/api/v1/user.ts:
```diff
@@ -8,7 +8,25 @@ const router = express.Router()
-router.use(express.json())
+router.use(express.text())
+
+/**
+ * Call para logar na api
+ */
+router.post('/login', async (req, res) => {
+	try {
+		req.body = JSON.parse(req.body)
+		if (!req.body.sessionId) throw 'Cookie Not Found' // verificação se existe no body o sessionId
+		req.user = await UserApi.findUser.byCookie(req.body.sessionId)
+		res.cookie('sessionId', req.body.sessionId) //reaplica o cookie sessionId para o subdominio
+		res.status(200).send({starus:'ok'})
+	} catch(err) {
+		res.status(401).send({
+			error: 'NotAuthorized',
+			message: 'A valid cookie \'sessionId\' needs to be informed in body to perform this operation'
+		})
+	}
+})
```
Por que tem uma função de login na API se já existe uma função de login?

GCastilho replied:
Eu acho que faz mais sentido a página de login fazer login na api, não em /login. O botão 'login' fazer um post pra /v1/user/authenticate ao invés de /login do main

Se isso facilitar o problema do cookie pode fazer, mas ele tem que ser httpOnly

GCastilho commented on src/server/api/v1/user.ts:
```diff
@@ -8,7 +8,25 @@ const router = express.Router()
 // Parsers
 router.use(cookieParser())
-router.use(express.json())
+router.use(express.text())
```
Por que?

GCastilho reviewed (commented):
Não é para alterar o funcionamento do servidor nesse commit, é para adicionar uma página de transações. As informações necessárias para isso estão na API http, NÃO É para criar entradas na API sem consultar eu ou o joão antes

Os problemas com autenticação eu vou resolver, mas fora isso as modificações não fazem sentido

Por favor, desfaça as modificações que vc fez ou fala pq cada uma delas foi necessária

GCastilho committed:
Corrigido subdomain no modo dev

-- Adicionado handler no sapper para que os subdomínios funcionem no modo dev

GCastilho committed:
Corrigido subdomain no modo dev

-- Adicionado handler no sapper para que os subdomínios funcionem no modo dev

GCastilho commented:
Eu já estou resolvendo os problemas com o servidor

GCastilho committed:
Revert "Criado serviço de roteamento para o modo DEV"

This reverts commit fb2ba4d251946e0b2d4111ff35454c98d3c96b85.

GCastilho committed:
Revert "Criado serviço de roteamento para o modo DEV"

This reverts commit fb2ba4d251946e0b2d4111ff35454c98d3c96b85.

Commit:
Fix e Implementações na lista

-- Retirado funções de /login e /lasttransaction da api
-- Adicionado request para transactions mais antigas passando o offset da lista
-- Conserto da transaction store para não criar operações diretas na variável da mesma

Commit:
Fix e Implementações na lista

-- Retirado funções de /login e /lasttransaction da api
-- Adicionado request para transactions mais antigas passando o offset da lista
-- Conserto da transaction store para não criar operações diretas na variável da mesma

Commit:
Error Handler na recepção de dados da api 

-- Add Informativo de erro na comunicação com a api
-- Retirado dados do CorewDraw dos icones
-- WIP layout dos itens da lista

Commit:
Error Handler na recepção de dados da api 

-- Add Informativo de erro na comunicação com a api
-- Retirado dados do CorewDraw dos icones
-- WIP layout dos itens da lista

Commit:
Carrega mais dados ao final da pagina

-- Alterado para carregar mais dados ao chegar ao final da pagina e não ao final da div

Commit:
Carrega mais dados ao final da pagina

-- Alterado para carregar mais dados ao chegar ao final da pagina e não ao final da div

Commit:
Alterado layout

-- Alterado carregamento de mais dados ao chegar no final (ultimos 100px da pagina)
-- Alterado distancia entre objetos (melhorias para modo Mobile)
-- Alterado filtros da tabela (retirado colunas adicionais)
-- Adicionado Dropdown para mais detalhes das transactions

Commit:
Alterado layout

-- Alterado carregamento de mais dados ao chegar no final (ultimos 100px da pagina)
-- Alterado distancia entre objetos (melhorias para modo Mobile)
-- Alterado filtros da tabela (retirado colunas adicionais)
-- Adicionado Dropdown para mais detalhes das transactions

Commit:
Fix duplicated Lines

-- Deletado linhas duplicadas

Commit:
Fix duplicated Lines

-- Deletado linhas duplicadas

Commit:
Merge branch 'master' into transactions-page

GCastilho committed:
Corrigido bug no master

-- Corrigido um bug que impedia o servidor de inicializar
-- Corrigido problemas do eslint

joaojvds committed:
Removido Lixo

-- Removido boa parte do código feito pelo TTO

joaojvds committed:
Merge remote-tracking branch 'origin/transactions-page' into transactions-page

joaojvds committed:
Implementado socket na store transaction

-- Implementado socket na store transaction

joaojvds committed:
implementado funções na store transactions

-- Ao fazer scroll no final da pagina mais transações serão carregadas
-- Refatorado sockets na store transactions

GCastilho committed:
Refatorado store de transactions

-- Refatorado os métodos da store de transactions sem afetar o funcionamento

GCastilho committed:
Refatorado o funcionamento da store de transactions

-- A store de transactions foi reprojetada e agora tem apenas um método público 'fetch' com um argumento opcional 'skip'
-- Corrigido algumas coisas na página de transactions

GCastilho committed:
Small fix

-- Melhorado o cálculo do skip

joaojvds committed:
Design preliminar

joaojvds committed:
Merge remote-tracking branch 'origin/transactions-page' into transactions-page

joaojvds committed:
Alterado estrutura da tabela

-- Por conta de limitações de como funcionam as tags table, foi trocado algumas delas para div e o seu display para grid

joaojvds committed:
Botão de cancelar(em desenvolvimento)

joaojvds committed:
Adicionado plugin svg

-- Adicionado o plugin para o cliente
-- Colocado o botão de cancelar na pagina
-- Alterado amount para number

joaojvds committed:
Pequenas mudanças na pagina transactions

joaojvds committed:
Eu acho que ta funcionando agora

joaojvds committed:
Alterações requisitadas pelo Gabriel

-- Separei os blocos reativos em dois
-- Uma correção de português
-- mudado algumas comparações para duplo igual

joaojvds committed:
Trocado a biblioteca de data

-- Trocado a biblioteca de data

joaojvds committed:
sameSite none

joaojvds committed:
Minor changes

-- Variavel de confirmações é setada para null caso o status for confirmado
-- Pequenas alterações no design

GCastilho committed:
API agora manda o fee da transalção

-- A API agora manda o fee de uma transaçãem uma propriedade fee

joaojvds committed:
Merge branch 'master' into transactions-page

joaojvds committed:
Mudança no tenario

-- Testa ai Gabriel

joaojvds committed:
Adicionado um afterUpdate

-- A variável confirmations é setada para null, quando o componente é atualizado

joaojvds committed:
Atualizado valor do fee mostrado na pagina

-- Agora a pagina mostra o fee que foi usado na transação

GCastilho committed:
Refatorado store de transactions

-- Refatora store de transactions
-- A store de transactions agora pede as próximas transações sem a necessidade de um parâmetro
-- A store de transactions agora se inicializa com as primeiras 10 transações automaticamente qdo o usuário faz login e reseta quando o usuário faz logout
-- Refatorado o handler de fim de página da página de transactions para de fato checar pelo fim da página, e não checar uma coisa aleatória da mente do joão
-- A página de transactions não mais faz um ataque DDoS quando o usuário faz scroll

GCastilho committed:
Refatorado store de transactions

-- Refatorado store de balances, ela agora não atualiza o saldo dela própria, mas permite uma atualização dele por outros módulos através de um método 'updateBalances'
-- Refatorado a store de transactions, ela agora atualiza o saldo da store de balances quando tem uma transação ou atualização de transação
-- Adicionado um método 'withdraw' na store de transactions para ser o método pelo qual o sistema faz uma nova transação. Esse método envia o request para a API e adiciona a transação ao array de transactions

GCastilho committed:
Resolvido problema da transação enviada não estar sendo atualizada na store

-- Resolvido problema onde uma transação recem enviada enstava sendo adicionada na store de maneira incorreta
-- Resolvido cor da tabela de transactions não reconhecer transação recem enviada como processing

joaojvds committed:
Consertado valores undefined

-- A data só é mostrada quando quando houver timestamp
-- Pequenas alterações na proporção da cédula de data

GCastilho committed:
Adicionado um método get ao store de currencies

-- A store de currencies agora tem um método get que retorna o objeto de uma currency requisitada
-- Corrigido um bug onde transações recebidas davam um TypeError pelo opid ser undefined
-- Removido módulo utils 'transactions' não mais utilizado

joaojvds committed:
Alterado ternário da data

GCastilho committed:
Bugfix no update de saldo

-- Corrigido um bug na atualização do saldo ao receber update de uma transação
-- O fee de uma transação enviada agora é enviado corretamente
-- Adicionado um bug para corrigir depois onde uma transação recebida não é atualizada corretamente
-- Minor fixes

joaojvds committed:
Criado uma tabela para modo mobile

-- Criado uma tabela para modo mobile

GCastilho committed:
Adicionado método de cancelar transações pendentes

-- Adicionado um método cancell na store de transactions para cancelar uma operação de saque ainda não executada
-- Corrigido o nome de algumas variáveis

joaojvds committed:
Implementado método no botão de cancelar

-- Agora o botão cancelar chama a função cancell na store transactions
-- Colocado uma desestruturação no objeto transaction

joaojvds committed:
Merge branch 'master' into transactions-page

GCastilho committed:
Mudado o momento onde a store de transactions é inicializada

-- A store de transactions agora não é mais inicializada no carregamento do site, mas sim qdo o usuário abre a página de transactions pela primeira vez
-- A store de transactions agora exporta uma variável para saber se a store está sincronizada

GCastilho committed:
Página de transações gora diferencia sem transações de carregando

-- A store de transactions agora exporta uma store 'synchronized', que indica se a store está sincronizada ou não
-- Adicionado um campo 'loading...' na página de transações que irá aparecer qdo não houver transações e a store não esteja sincronizada
-- Se o cliente receber um update de uma transação que não está na store, a store irá tentar buscar essa transação usando a API e adicionar ela na posição correta da lista
-- Minor fixes

GCastilho committed:
Adicionado handlers para as respostas do cancell_pending

-- Adicionado handler para a resposta do servidor 'cancelled' que indica que a transação foi imediatamente cancelada
-- Adicionado handler para o evento de update_sent_tx com status cancelled

GCastilho committed:
Merge branch 'master' into transactions-page

gabr1gus committed:
improvements and bug fix

--corrigido erro que uma transaçao nao era cancelada se ela ja havia sido removido do checklist de transaçoes pendentes
--adicionado uma detecçao e resposta para se a transaçao ja foi efetuada quando se tenta cancelar

gabr1gus committed:
minor fix

--adicionado check pra evitar o retorno de error 11000 desnecessários

gabr1gus committed:
trocado a tipagem de message

--agora recebe o tipo correto de response para o request de cancell

gabr1gus committed:
Corrigido erro de tratamento de lixo no cancell withdraw loop

-- Withdraw loop teria um fim prematuro se tivesse algum lixo no meio da checklist de cancel withdraw
-- o erro foi corrigido tratando o lixo como completo, que vai depois ser pego pelo garbage collector

gabr1gus committed:
Merge branch 'cancell-withdraw-fix' into transactions-page

joaojvds committed:
Refatorado tabela do modo mobile

-- Alterado CSS da pagina de transactions
-- Colocado animação na tabela mobile

joaojvds committed:
Merge remote-tracking branch 'origin/transactions-page' into transactions-page

joaojvds committed:
Alterado ícone do header 

-- Alterado ícone do botão transactions no dropdown menu

joaojvds committed:
Alterado ícone da pagina de transaction 

-- Mudado texto na pagina change password
-- Centralizado título da pagina transaction

GCastilho committed:
Corrigido um erro de formatação

GCastilho reviewed (approved):
Mais de 5 meses...
